### DISCUSSION TITLE
GLib-ERROR

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#2303] [Q&A] [answered]
Hello, in an unpredictable manner I'm getting the following message and Cooja aborts. Can anyone help me understand what can I do?

[java] (java:1666): GLib-ERROR **: 16:34:56.269: ../../../../glib/gmem.c:135: failed to allocate 134217728 bytes
     [java] Java Result: 133

The 134217728 may be indicative because it is 128MB but I don't know who or when is asking for that memory.

Thanks,
Rahav

**Comment by pjonsson** [comment, marked as answer]
I have no idea where the message comes from (or what is trying to allocate 128M), but there is some kind of bigmem profile in the ant build file (build.xml) that you could try using.

If you're talking about v4.7, I think the maxmem is 512M, so if you still run out of memory, it might be possible to bump that to 1024M or even 2048M. Do not go past 2048M, the v4.8 Cooja does not support that. I have no idea if the v4.7 Cooja supports 2048M.

**Comment by rahav** [comment]
I didn't find the source of the problem but I think it occurs in Cooja. Anecdotal evidence suggests this occurs from often when Cooja is Paused and Started a few times during a sim.

**Reply by pjonsson** [reply]
There were race conditions with starting/stopping the simulation in the GUI until recently (past v4.8), so you could be hitting one of those. Waiting a little bit (~1 second?) before starting again after stopping the simulation might avoid the issue sometimes. Cooja for v4.8 should still be an improvement over v4.7.

Did the issue get less frequent with a bigmem profile?
